src/dataregistry/registrar.py

```diff
@@ -344,7 +350,11 @@ def register_dataset(self, relative_path, version,
         values["version_patch"] = v_fields["patch"]
         values["version_string"] = version_string
         if version_suffix: values["version_suffix"] = version_suffix
-        if creation_date: values["dataset_creation_date"] = creation_date
+        if creation_date:
+            values["dataset_creation_date"] = creation_date
+        else:
+            if ds_creation_date:
```

Comment:
Is there any reason why `ds_creation_date` would not be set, does it not have to exist else it would throw an error in handle data?

Then `if ds_creation_date` would not be needed

Reply:
`ds_creation_date`  will be set if either of the following is true:
 1. Caller has set it with `creation_date` optional argument
 2. Dataset is not a dummy, in which case the value found by `_handle_data` if there isn't already a value from 1.
 
 But if neither of those holds `ds_creation_date` is None.

Reply:
thanks!